test-plan/eventing/error-signalling-conformance.md

```diff
@@ -0,0 +1,382 @@
+VERSION:  messaging.knative.dev/v1
+
+RESOURCE: conditions <[]Object>
+
+[...]
+
+FIELDS:
+   lastTransitionTime   <string>
+     LastTransitionTime is the last time the condition transitioned from one
+     status to another. We use VolatileTime in place of metav1.Time to exclude
+     this from creating equality.Semantic differences (all other things held
+     constant).
+
+   message      <string>
+     A human readable message indicating details about the transition.
+
+   reason       <string>
+     The reason for the condition's last transition.
+
+   severity     <string>
+     Severity with which to treat failures of this type of condition. When this
+     is not specified, it defaults to Error.
+
+   status       <string> -required-
+     Status of the condition, one of True, False, Unknown.
+
+   type <string> -required-
+     Type of condition.
+```
+
+---
+
+## [Test] OpenAPI schema for the Subscription kind
+
+Describe the registered `subscriptions.messaging.knative.dev` kind and verify
+that it contains a `status.conditions` field, expressed as a list of objects
+which themselves exclusively contain the fields:
+
+- `type`, typed as a string and marked as required
+- `status`, typed as a string
+- `message` typed as a string
+- `severity`, typed as a string
+- `lastTransitionTime`, typed as a string
+
+Explain the `status` field:
+
+```
+kubectl explain subscriptions.messaging.knative.dev.status.conditions
+```
+
+### [Output]
+
+```
+KIND:     Subscription
+VERSION:  messaging.knative.dev/v1
+
+RESOURCE: conditions <[]Object>
+
+[...]
+
+FIELDS:
+   lastTransitionTime   <string>
+     LastTransitionTime is the last time the condition transitioned from one
+     status to another. We use VolatileTime in place of metav1.Time to exclude
+     this from creating equality.Semantic differences (all other things held
+     constant).
+
+   message      <string>
+     A human readable message indicating details about the transition.
+
+   reason       <string>
+     The reason for the condition's last transition.
+
+   severity     <string>
+     Severity with which to treat failures of this type of condition. When this
+     is not specified, it defaults to Error.
+
+   status       <string> -required-
+     Status of the condition, one of True, False, Unknown.
+
+   type <string> -required-
+     Type of condition.
+```
+
+---
+
+## [Pre] Create a Broker instance which reconciliation fails
+
+```
+kubectl create -f control-plane/error-signalling/broker-not-ready.yaml
+```
+
+## [Test] Broker failure is propagated to the Ready status condition
+
+The failure caused by the intentionally invalid spec is propagated to the `Ready` condition, which has a `status` of
+`False`, a severity of `""` (error), and a non-empty `reason`:
+
+```
+kubectl get brokers/not-ready -o jsonpath='{.status.conditions[?(@.type=="Ready")].status}{";"}{.status.conditions[?(@.type=="Ready")].severity}{";"}{.status.conditions[?(@.type=="Ready")].reason}'
+```
+
```

Comment:
Here you can search if this is being tested on knative/eventing and put a link to it (search in the test/rekt folder):
[Here](https://github.com/knative/specs/pull/67/files#diff-91e25876f53e60df73f07792825e96a63a4e5adc5a6d2069e29f19018883d026R58) and [Here](https://github.com/knative/specs/pull/67/files#diff-91e25876f53e60df73f07792825e96a63a4e5adc5a6d2069e29f19018883d026R81)

Reply:
Same for all the tests you think are valuable or testable. If you dont found any place where they are been tested just put it here

Reply:
Will do